audio/out/ao_audiotrack.c

```diff
@@ -524,26 +527,37 @@ static int AudioTrack_write(struct ao *ao, int len)
 
 static void uninit_jni(struct ao *ao)
 {
-    JNIEnv *env = MP_JNI_GET_ENV(ao);
-    for (int i = 0; i < MP_ARRAY_SIZE(jclass_list); i++) {
-        mp_jni_reset_jfields(env, jclass_list[i].fields,
-                             jclass_list[i].mapping, 1, ao->log);
+    mp_mutex_lock(&jni_static_lock);
+    jni_static_use_count--;
+    if (jni_static_use_count == 0) {
+        JNIEnv *env = MP_JNI_GET_ENV(ao);
+        for (int i = 0; i < MP_ARRAY_SIZE(jclass_list); i++) {
+            mp_jni_reset_jfields(env, jclass_list[i].fields,
+                                jclass_list[i].mapping, 1, ao->log);
+        }
     }
+    mp_mutex_unlock(&jni_static_lock);
 }
 
 static int init_jni(struct ao *ao)
 {
-    JNIEnv *env = MP_JNI_GET_ENV(ao);
-    for (int i = 0; i < MP_ARRAY_SIZE(jclass_list); i++) {
-        if (mp_jni_init_jfields(env, jclass_list[i].fields,
-                                jclass_list[i].mapping, 1, ao->log) < 0) {
-            goto error;
+    mp_mutex_lock(&jni_static_lock);
+    if (jni_static_use_count == 0) {
+        JNIEnv *env = MP_JNI_GET_ENV(ao);
+        for (int i = 0; i < MP_ARRAY_SIZE(jclass_list); i++) {
+            if (mp_jni_init_jfields(env, jclass_list[i].fields,
+                                    jclass_list[i].mapping, 1, ao->log) < 0) {
+                goto error;
+            }
         }
     }
+    jni_static_use_count++;
+    mp_mutex_unlock(&jni_static_lock);
     return 0;
 
 error:
     uninit_jni(ao);
+    mp_mutex_unlock(&jni_static_lock);
```

Comment:
deadlock here because `uninit_jni` also tries to lock

Reply:
Yes, I'll fix this.